File: LeetCode/680. 验证回文字符串 Ⅱ.cpp
class Solution {
private:
    bool isPalindrome(string &s,int l,int r){
        for(;l<r&&s[l]==s[r];l++,r--);
        return l>=r;
    }
public:
    bool validPalindrome(string s) {
        int l=0,r=s.size()-1;
        for(;l<r&&s[l]==s[r];l++,r--);
        return isPalindrome(s,l+1,r)||isPalindrome(s,l,r-1);
    }
};

File: LeetCode/680. 验证回文字符串 Ⅱ.rs
impl Solution {
  pub fn valid_palindrome(s: String) -> bool {
      let s=s.as_bytes();
      let (mut l,mut r)=(0,s.len()-1);
      let is_palindrome=|mut l,mut r|{
          while l<r&&s[l]==s[r]{
              l+=1;
              r-=1;
          }
          return l>=r;
      };
      while l<r&&s[l]==s[r]{
          l+=1;
          r-=1;
      }
      return is_palindrome(l+1,r)||is_palindrome(l,r-1);
  }
}